go-controller/pkg/util/net_linux.go

```diff
@@ -392,6 +392,11 @@ func DeleteConntrack(ip string, port int32, protocol kapi.Protocol) error {
 		if err := filter.AddProtocol(132); err != nil {
 			return fmt.Errorf("could not add Protocol SCTP to conntrack filter %v", err)
 		}
+	} else if protocol == kapi.ProtocolTCP {
```

Comment:
I'm not sure if we want to add TCP. Especially we dont want to add it into this commit. It should be a separate commit. TCP doesn't need to be flushed because its connection oriented and it has TIME_WAIT state for ensuring connections are closed cleanly. By flushing all the CT entries in TCP, you could create errors in applications that are expecting the right TCP connection shutdown, as well as potentially introduce errors in subsequent TCP connections. This is a really good article describing some scenarios:
http://www.serverframework.com/asynchronousevents/2011/01/time-wait-and-its-design-implications-for-protocols-and-scalable-servers.html

Unless there is a bug specific to TCP conntrack state handling with ovnk I wouldn't touch this.

Reply:
Additionally I'm not sure why we are flushing SCTP here as that is connection oriented as well.

Reply:
problem is this bug is not protocol specific, this specific bug happens in SCTP when a LB service is re-created in the middle of an active ESTABLISHED connection where stream is being sent and suddenly the clusterIP changes. Is it possible for us to hit this in TCP keep-alive setups or will TIME_WAIT protect us?

Note that even in SCTP once the SHUTDOWN, SHUTDOWN_ACK & SHUTDOWN COMPLETE conntrack removes the entry, this is a corner case where the connections are active towards the old service and they recreate service in the middle which is why I thought it could happen for TCP/UDP as well.

either ways I need to either add TCP here as separate commit or do a check before calling the deleteConntrack function to make sure we don't fail for the unprotected TCP case.

Reply:
I'd say its more of a consequence of that LB->CIP & NP->CIP DNAT we do :)

Reply:
based on our upstream conversation I think we will keep TCP and SCTP and UDP for service deletion and for endpoints only UDP.